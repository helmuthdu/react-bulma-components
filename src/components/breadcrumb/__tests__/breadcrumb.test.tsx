import { render } from '@testing-library/react';
import * as React from 'react';
import { Breadcrumb } from '..';

describe('Breadcrumb component', () => {
  it('should render', () => {
    const { asFragment } = render(<Breadcrumb />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render list of items', () => {
    const { asFragment } = render(
      <Breadcrumb
        items={[
          {
            url: '/',
            name: 'Home'
          },
          {
            url: '/section',
            name: 'Section'
          },
          {
            url: '/detail',
            name: 'Details',
            active: true
          }
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  ['arrow', 'dot', 'bullet', 'succeeds'].map(separator =>
    it(`should use separator ${separator}`, () => {
      const { asFragment } = render(
        <Breadcrumb
          separator={separator as any}
          items={[
            {
              name: 'Storybook',
              url: '#1'
            },
            {
              name: 'Breadcrumb',
              url: '#2'
            },
            {
              name: 'Breadcrumb Types',
              url: '#3',
              active: true
            }
          ]}
        />
      );
      expect(asFragment()).toMatchSnapshot();
    })
  );

  it('should use inline style and custom size', () => {
    const { asFragment } = render(
      <Breadcrumb
        style={{ marginTop: 10 }}
        size="large"
        items={[
          {
            url: '/',
            name: 'Home'
          },
          {
            url: '/section',
            name: 'Section'
          },
          {
            url: '/detail',
            name: 'Details',
            active: true
          }
        ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
