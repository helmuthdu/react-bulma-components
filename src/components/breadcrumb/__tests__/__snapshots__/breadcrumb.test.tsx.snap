// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component should render 1`] = `
<DocumentFragment>
  <nav
    class="breadcrumb"
  >
    <ul />
  </nav>
</DocumentFragment>
`;

exports[`Breadcrumb component should render list of items 1`] = `
<DocumentFragment>
  <nav
    class="breadcrumb"
  >
    <ul>
      <li
        class=""
      >
        <a>
          Home
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Section
        </a>
      </li>
      <li
        class="is-active"
      >
        <a>
          Details
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Breadcrumb component should use inline style and custom size 1`] = `
<DocumentFragment>
  <nav
    class="breadcrumb is-large"
    style="margin-top: 10px;"
  >
    <ul>
      <li
        class=""
      >
        <a>
          Home
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Section
        </a>
      </li>
      <li
        class="is-active"
      >
        <a>
          Details
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Breadcrumb component should use separator arrow 1`] = `
<DocumentFragment>
  <nav
    class="breadcrumb has-arrow-separator"
  >
    <ul>
      <li
        class=""
      >
        <a>
          Storybook
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Breadcrumb
        </a>
      </li>
      <li
        class="is-active"
      >
        <a>
          Breadcrumb Types
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Breadcrumb component should use separator bullet 1`] = `
<DocumentFragment>
  <nav
    class="breadcrumb has-bullet-separator"
  >
    <ul>
      <li
        class=""
      >
        <a>
          Storybook
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Breadcrumb
        </a>
      </li>
      <li
        class="is-active"
      >
        <a>
          Breadcrumb Types
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Breadcrumb component should use separator dot 1`] = `
<DocumentFragment>
  <nav
    class="breadcrumb has-dot-separator"
  >
    <ul>
      <li
        class=""
      >
        <a>
          Storybook
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Breadcrumb
        </a>
      </li>
      <li
        class="is-active"
      >
        <a>
          Breadcrumb Types
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Breadcrumb component should use separator succeeds 1`] = `
<DocumentFragment>
  <nav
    class="breadcrumb has-succeeds-separator"
  >
    <ul>
      <li
        class=""
      >
        <a>
          Storybook
        </a>
      </li>
      <li
        class=""
      >
        <a>
          Breadcrumb
        </a>
      </li>
      <li
        class="is-active"
      >
        <a>
          Breadcrumb Types
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
